### PR TITLE
6244- Datagrid: Leading spaces are visible upon blur 

### PR DESCRIPTION
**Pull request by sesbrenosienna**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR adds a trimSpaces option for leading spaces upon blur. 

**Related github/jira issue (required)**:
Closes https://github.com/infor-design/enterprise/issues/6244

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app
- Go to http://localhost:4000/components/datagrid/example-editable.html
- Click on the "Assemble Paint" cell 
- Add leading Spaces 
- Click away from the cell
- Click on the "Assemble Paint" cell 

**Included in this Pull Request**:
- [ ] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

**Comment by s-werking** [comment]
Hi @sesbrenosienna, any reason that this setting was added to the toolbar settings of a datagrid and not a datagrid setting on its own?

File: src/components/datagrid/datagrid.js [hunk 1]
```diff
@@ -10870,8 +10870,12 @@ Datagrid.prototype = {
 
     // Adjust leading/trailing spaces as `&nbsp;`
     const adj = (thisVal, regx) => {
-      const r = (typeof thisVal === 'string' ? thisVal.match(regx) : ['']) || [''];
-      return r[0].replace(/\s/g, '&nbsp;');
+      let r = (typeof thisVal === 'string' ? thisVal.match(regx) : ['']) || [''];
+
+      if (!this.settings.toolbar.trimSpaces) {
```
Comment (tmcconechy) [inline src/components/datagrid/datagrid.js:10875]: Agreed with @s-werking this setting should be on the datagrid not toolbar as its not toolbar related

```
this.settings.trimSpaces
```
Also can you add to the top of the file so it appears in the Docs?

Comment (tmcconechy) [reply src/components/datagrid/datagrid.js:10875]: Thanks @sesbrenosienna otherwise works good

**Review by yohannahbautista — APPROVED** [review]

**Review by ericangeles — APPROVED** [review]
Nice! 🍌

**Pull request merged**